### ISSUE TITLE
`the egui glow has already been destroyed` panic in eframe

### ISSUE THREAD

**Issue by Zoxc** [emilk/egui#2027]
I get the following panic when closing an eframe window on Windows:
```
thread 'main' panicked at 'the egui glow has already been destroyed!', B:\Programmering\egui\egui\crates\egui_glow\src\painter.rs:699:9
```

**Comment by emilk** [comment]
This should only happen if you are somehow accessing the glow painter after shutting down the app (I don't see how).

Please run with `RUST_BACKTRACE=1` to see how this happened.

Also: does this happen on latest `master`?

**Comment by Zoxc** [comment]
It started happening after https://github.com/emilk/egui/pull/2013.

```
thread 'main' panicked at 'the egui glow has already been destroyed!', crates\egui_glow\src\painter.rs:699:9
stack backtrace:
   0: std::panicking::begin_panic_handler
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e\/library\std\src\panicking.rs:584
   1: core::panicking::panic_fmt
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e\/library\core\src\panicking.rs:143
   2: egui_glow::painter::Painter::assert_not_destroyed
             at .\crates\egui_glow\src\painter.rs:699
   3: egui_glow::painter::Painter::paint_primitives
             at .\crates\egui_glow\src\painter.rs:376
   4: egui_glow::painter::Painter::paint_and_update_textures
             at .\crates\egui_glow\src\painter.rs:342
   5: eframe::native::run::glow_integration::impl$1::paint
             at .\crates\eframe\src\native\run.rs:457
   6: eframe::native::run::run_and_return::closure$0<eframe::native::run::glow_integration::GlowWinitApp>
             at .\crates\eframe\src\native\run.rs:96
   7: winit::platform_impl::platform::event_loop::impl$3::run_return::closure$0<eframe::native::run::RequestRepaintEvent,eframe::native::run::run_and_return::closure_env$0<eframe::native::run::glow_integration::GlowWinitApp> >
             at C:\Users\John\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.27.2\src\platform_impl\windows\event_loop.rs:254
   8: alloc::boxed::impl$45::call_mut<tuple$<enum$<winit::event::Event<eframe::native::run::RequestRepaintEvent> >,ref_mut$<enum$<winit::event_loop::ControlFlow> > >,dyn$<core::ops::function::FnMut<tuple$<enum$<winit::event::Event<eframe::native::run::RequestRe
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e\library\alloc\src\boxed.rs:1868
   9: winit::platform_impl::platform::event_loop::runner::impl$3::call_event_handler::closure$0<eframe::native::run::RequestRepaintEvent>
             at C:\Users\John\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.27.2\src\platform_impl\windows\event_loop\runner.rs:245
  10: core::panic::unwind_safe::impl$23::call_once<tuple$<>,winit::platform_impl::platform::event_loop::runner::impl$3::call_event_handler::closure_env$0<eframe::native::run::RequestRepaintEvent> >
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e\library\core\src\panic\unwind_safe.rs:271
  11: std::panicking::try::do_call<core::panic::unwind_safe::AssertUnwindSafe<winit::platform_impl::platform::event_loop::runner::impl$3::call_event_handler::closure_env$0<eframe::native::run::RequestRepaintEvent> >,tuple$<> >
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e\library\std\src\panicking.rs:492
  12: std::panicking::try::do_catch<core::panic::unwind_safe::AssertUnwindSafe<winit::platform_impl::platform::event_loop::public_window_callback_inner::closure_env$0<eframe::native::run::RequestRepaintEvent> >,isize>
  13: std::panicking::try<tuple$<>,core::panic::unwind_safe::AssertUnwindSafe<winit::platform_impl::platform::event_loop::runner::impl$3::call_event_handler::closure_env$0<eframe::native::run::RequestRepaintEvent> > >
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e\library\std\src\panicking.rs:456
  14: std::panic::catch_unwind<core::panic::unwind_safe::AssertUnwindSafe<winit::platform_impl::platform::event_loop::runner::impl$3::call_event_handler::closure_env$0<eframe::native::run::RequestRepaintEvent> >,tuple$<> >
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e\library\std\src\panic.rs:137
  15: winit::platform_impl::platform::event_loop::runner::EventLoopRunner<eframe::native::run::RequestRepaintEvent>::catch_unwind<eframe::native::run::RequestRepaintEvent,tuple$<>,winit::platform_impl::platform::event_loop::runner::impl$3::call_event_handler::c
             at C:\Users\John\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.27.2\src\platform_impl\windows\event_loop\runner.rs:154
  16: winit::platform_impl::platform::event_loop::runner::EventLoopRunner<eframe::native::run::RequestRepaintEvent>::call_event_handler<eframe::native::run::RequestRepaintEvent>
             at C:\Users\John\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.27.2\src\platform_impl\windows\event_loop\runner.rs:239
  17: winit::platform_impl::platform::event_loop::runner::EventLoopRunner<eframe::native::run::RequestRepaintEvent>::call_redraw_events_cleared<eframe::native::run::RequestRepaintEvent>
             at C:\Users\John\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.27.2\src\platform_impl\windows\event_loop\runner.rs:406
  18: winit::platform_impl::platform::event_loop::runner::EventLoopRunner<eframe::native::run::RequestRepaintEvent>::move_state_to<eframe::native::run::RequestRepaintEvent>
             at C:\Users\John\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.27.2\src\platform_impl\windows\event_loop\runner.rs:355
  19: winit::platform_impl::platform::event_loop::runner::EventLoopRunner<eframe::native::run::RequestRepaintEvent>::redraw_events_cleared<eframe::native::run::RequestRepaintEvent>
             at C:\Users\John\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.27.2\src\platform_impl\windows\event_loop\runner.rs:231
  20: winit::platform_impl::platform::event_loop::thread_event_target_callback::closure$0<eframe::native::run::RequestRepaintEvent>
             at C:\Users\John\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.27.2\src\platform_impl\windows\event_loop.rs:2260
  21: core::ops::function::FnOnce::call_once<winit::platform_impl::platform::event_loop::thread_event_target_callback::closure_env$0<eframe::native::run::RequestRepaintEvent>,tuple$<> >
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e\library\core\src\ops\function.rs:227
  22: core::panic::unwind_safe::impl$23::call_once<isize,winit::platform_impl::platform::event_loop::thread_event_target_callback::closure_env$0<eframe::native::run::RequestRepaintEvent> >
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e\library\core\src\panic\unwind_safe.rs:271
  23: std::panicking::try::do_call<core::panic::unwind_safe::AssertUnwindSafe<winit::platform_impl::platform::event_loop::thread_event_target_callback::closure_env$0<eframe::native::run::RequestRepaintEvent> >,isize>
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e\library\std\src\panicking.rs:492
  24: std::panicking::try::do_catch<core::panic::unwind_safe::AssertUnwindSafe<winit::platform_impl::platform::event_loop::public_window_callback_inner::closure_env$0<eframe::native::run::RequestRepaintEvent> >,isize>
  25: std::panicking::try<isize,core::panic::unwind_safe::AssertUnwindSafe<winit::platform_impl::platform::event_loop::thread_event_target_callback::closure_env$0<eframe::native::run::RequestRepaintEvent> > >
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e\library\std\src\panicking.rs:456
  26: std::panic::catch_unwind<core::panic::unwind_safe::AssertUnwindSafe<winit::platform_impl::platform::event_loop::thread_event_target_callback::closure_env$0<eframe::native::run::RequestRepaintEvent> >,isize>
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e\library\std\src\panic.rs:137
  27: winit::platform_impl::platform::event_loop::runner::EventLoopRunner<eframe::native::run::RequestRepaintEvent>::catch_unwind<eframe::native::run::RequestRepaintEvent,isize,winit::platform_impl::platform::event_loop::thread_event_target_callback::closure_en
             at C:\Users\John\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.27.2\src\platform_impl\windows\event_loop\runner.rs:154
  28: winit::platform_impl::platform::event_loop::thread_event_target_callback<eframe::native::run::RequestRepaintEvent>
             at C:\Users\John\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.27.2\src\platform_impl\windows\event_loop.rs:2436
  29: CallWindowProcW
  30: DispatchMessageW
  31: SendMessageTimeoutW
  32: KiUserCallbackDispatcher
  33: NtUserDispatchMessage
  34: DispatchMessageW
  35: winit::platform_impl::platform::event_loop::EventLoop<eframe::native::run::RequestRepaintEvent>::run_return<eframe::native::run::RequestRepaintEvent,eframe::native::run::run_and_return::closure_env$0<eframe::native::run::glow_integration::GlowWinitApp> >
             at C:\Users\John\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.27.2\src\platform_impl\windows\event_loop.rs:276
  36: winit::platform::run_return::impl$0::run_return<eframe::native::run::RequestRepaintEvent,eframe::native::run::run_and_return::closure_env$0<eframe::native::run::glow_integration::GlowWinitApp> >
             at C:\Users\John\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.27.2\src\platform\run_return.rs:62
  37: eframe::native::run::run_and_return<eframe::native::run::glow_integration::GlowWinitApp>
             at .\crates\eframe\src\native\run.rs:84
  38: eframe::native::run::glow_integration::run_glow::closure$0
             at .\crates\eframe\src\native\run.rs:592
  39: eframe::native::run::with_event_loop::closure$0<eframe::native::run::glow_integration::run_glow::closure_env$0>
             at .\crates\eframe\src\native\run.rs:73
  40: std::thread::local::LocalKey<core::cell::RefCell<enum$<core::option::Option<winit::event_loop::EventLoop<eframe::native::run::RequestRepaintEvent> >, 0, 3, Some> > >::try_with<core::cell::RefCell<enum$<core::option::Option<winit::event_loop::EventLoop<efr
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e\library\std\src\thread\local.rs:442
  41: std::thread::local::LocalKey<core::cell::RefCell<enum$<core::option::Option<winit::event_loop::EventLoop<eframe::native::run::RequestRepaintEvent> >, 0, 3, Some> > >::with<core::cell::RefCell<enum$<core::option::Option<winit::event_loop::EventLoop<eframe:
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e\library\std\src\thread\local.rs:418
  42: eframe::native::run::with_event_loop<eframe::native::run::glow_integration::run_glow::closure_env$0>
             at .\crates\eframe\src\native\run.rs:66
  43: eframe::native::run::glow_integration::run_glow
             at .\crates\eframe\src\native\run.rs:589
  44: eframe::run_native
             at .\crates\eframe\src\lib.rs:172
  45: egui_demo_app::main
             at .\crates\egui_demo_app\src\main.rs:20
  46: core::ops::function::FnOnce::call_once<void (*)(),tuple$<> >
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e\library\core\src\ops\function.rs:227
```

**Comment by emilk** [comment]
Can you give me an exact repro? e.g. "Run `cargo run -p hello_world` and click the X on the window"

**Comment by emilk** [comment]
Also: run with `RUST_LOG=trace` for more details on what happens. Or run with a debugger!

**Comment by emilk** [comment]
Does https://github.com/emilk/egui/commit/e0c7533ede56c770b317f77597990ca5ab8048e0 help?

**Comment by ar37-rs** [comment]
fixes on #2030

**Comment by ar37-rs** [comment]
@emilk Yup, fixes on [`e0c7553`](https://github.com/emilk/egui/commit/e0c7533ede56c770b317f77597990ca5ab8048e0)
